qiskit/transpiler/passes/optimization/commutative_inverse_cancellation.py

```diff
@@ -36,9 +54,31 @@ def _skip_node(self, node):
             return True
         if node.op.is_parameterized():
             return True
-        # ToDo: possibly also skip nodes on too many qubits
         return False
 
+    def _inverse_upto_phase(self, node1, node2):
```

Comment:
Is this more accurate?
```suggestion
    def _check_matrix_equivalence(self, node1, node2):
```

Reply:
Hmm, this function checks whether the two nodes are inverses of each other, so using the word `equivalence` sounds a bit misleading. And only does the matrix check when the corresponding option is set. I have changed it to `_check_inverse`, hopefully it's a bit better. Anyway, it's a private function and we can change its name freely on every release :).